### ISSUE TITLE
[Lecture Topic Task] Documenting Stategy Pattern (Section 5.2)

### ISSUE THREAD

**Issue by JConcepcion01** [uprm-inso4116-2024-2025-s1/semester-project-trolley-tracker-app#176]
Please refer to the following document for more details: https://docs.google.com/document/d/1iqslezp9a2G_6m-8B7yVRagj3M7xjivETaEn-bo2t2M/edit?usp=sharing)

It is very important that you know that these tasks require more than just writing. You need to make sure that such features you are writing about are implemented on the code/project itself and provide proof, a screenshot or something for it. If it is not implemented which most likely most of them wont, you’d need to develop them yourself on the code. Sections have already been added on the documentation with the professor’s notes for it, you just need to fill it.

These tasks are basically Developing + Documentation + Lecture Topic at the same time since they are class topic related, you need to develop them if they’re not and document them. So plan accordingly!

**Comment by JoelTorres20** [comment]
We have implemented the Strategy Pattern to modularize and simplify the logic for toggling different menus in the Trolley Tracker Admin Dashboard. The Strategy Pattern allows dynamic selection of menu-specific behaviors at runtime, reducing the coupling between components and improving code maintainability.

Example Use Case: Menu Toggles

The Trolley Tracker Admin Dashboard provides multiple side menus (e.g., Active Trips, Notifications) for user interaction. Each menu requires unique behavior, such as toggling visibility and managing states. The Strategy Pattern allows us to encapsulate these behaviors into individual strategy classes, making the system more flexible and extensible.

Implementation Details

Strategy Interface:
- We created a shared interface, MenuStrategy, which defines a common toggle  method. This ensures all menu strategies adhere to the same contract.

Concrete Strategies:

- Implementations of MenuStrategy were created for specific menu behaviors:
  - LeftMenuStrategy: Handles toggling the left-side navigation menu.
  - ActiveTripsMenuStrategy: Manages the behavior for toggling the Active Trips menu.
  - NotificationsMenuStrategy: Manages the behavior for toggling the Notifications menu.
  
Integration with NavbarService:
- The NavbarService class was refactored to utilize the Strategy Pattern. It dynamically selects a menu strategy at runtime based on the context or user interaction.

Component-Level Integration:
- The NavbarComponent interacts with the NavbarService to set the appropriate strategy and trigger the toggleMenu method.
- For example, clicking the "Notifications" button sets the NotificationsMenuStrategy before toggling the menu.

Code Example
The refactored and added code can be viewed here: [[GitHub Gist](https://gist.github.com/JoelTorres20/28e42454df6212c944bd6e373a4d9f7b)]

Also the pull request for this task can be accessed [here](https://github.com/uprm-inso4116-2024-2025-s1/RUMBus-Dashboard/pull/38)
